.travis.yml

```diff
@@ -1,16 +1,10 @@
-before_install:
-  - nvm install-latest-npm
 script:
   - npm run build
+  - npm run build-spec
+  - if [ ! -z "$(git status --porcelain)" ]; then echo "repo is dirty; you probably need to 'npm run build-spec' and commit the result"; exit 1; fi
   - npm test
   - npm run lint
   - npm run test-published-files
```

Comment:
this is a lot of things to run in serial, they should probably be split into separate jobs soon (doesn't have to be in this PR)